pkg/controllers/machinesetsync/machineset_sync_controller.go

```diff
@@ -1399,7 +1400,13 @@ func compareCAPIMachineSets(capiMachineSet1, capiMachineSet2 *clusterv1.MachineS
 		diff[".spec"] = diffSpec
 	}
 
-	if diffObjectMeta := util.ObjectMetaEqual(capiMachineSet1.ObjectMeta, capiMachineSet2.ObjectMeta); len(diffObjectMeta) > 0 {
+	diffCAPIMachineSet1 := capiMachineSet1.DeepCopy()
+	delete(diffCAPIMachineSet1.Annotations, "cluster.x-k8s.io/conversion-data")
+
+	diffCAPIMachineSet2 := capiMachineSet1.DeepCopy()
+	delete(diffCAPIMachineSet2.Annotations, "cluster.x-k8s.io/conversion-data")
```

Comment:
I'd move these into https://github.com/openshift/cluster-capi-operator/blob/a1a97d62e7847436ff8b60843b79536e3f0cc517/pkg/util/sync.go#L56-L58

we explicitly say there that's only including the fields/structs we care about for the MAPI-CAPI conversion.

Also independently on whether we add them there or not I'd add a comment on why we are dropping these annotations :)

Reply:
My thought was: the util func is also used for non-capi objects. But thinking again about it: totally fine to add this directly to the util 👍 

Updated and also added a description inside ObjectMetaEqual.